app/models/user.rb

```diff
@@ -56,21 +56,41 @@
 class User < ApplicationRecord
   devise :database_authenticatable, :lockable, :validatable, :timeoutable, :trackable, :invitable, :recoverable
 
-  belongs_to :vita_partner, optional: true
   has_many :assigned_tax_returns, class_name: "TaxReturn", foreign_key: :assigned_user_id
+
+  #
+  belongs_to :vita_partner, optional: true
   has_and_belongs_to_many :supported_organizations,
            join_table: "users_vita_partners",
            class_name: "VitaPartner"
+  #
+
+  has_many :memberships
```

Comment:
I think it would be really helpful to have a many many relation defined that allows us to easily query vita_partners through the memberships table.

```ruby
has_many :vita_partners, through: :memberships
```
Should a user truly have multiple memberships though? 

If this replaces `User#vita_partner`, then how will we know which membership is the "primary" one, the org this user belongs to?

- If we are replacing vita_partner_id with a :member role, should we enforce that a user has one and only one "role: member" membership?
- Should we also enforce that the orgs they are "leads" for are not the same as their "primary" organization that they are member for?
  - Or can you be both a lead and member of the same org?
  - it seems implicit that the org you belong to is the one you are a "member" for. Should we separate that more explicitly from other memberships?

Should this have uniqueness constraints or can a user have two memberships with the same role for the same organization?

Reply:
There's no existing ticket or use case that requires a designation of "primary" vita partner, but should a story require that in the future we can add a is_primary field to the membership and ensure that only one membership can be marked as primary.